client/scripts/controllers/server/socket/chainEventsNs.ts

```diff
@@ -0,0 +1,59 @@
+import { WebsocketMessageType, WebsocketNamespaces } from 'types';
+import app from 'state';
+import { Notification, NotificationSubscription } from 'models';
+import { io } from 'socket.io-client';
+
+
+export class ChainEventsNamespace {
+	private ceNs;
+	private _isConnected = false;
+
+	constructor() {
+		this.ceNs = io(`/${WebsocketNamespaces.ChainEvents}`, {transports: ['websocket']});
+		this.ceNs.on('connect', this.onconnect.bind(this));
+		this.ceNs.on('disconnect', this.ondisconnect.bind(this))
+		this.ceNs.on(WebsocketMessageType.ChainEventNotification, this.onChainEvent.bind(this))
+	}
+
+	public addChainEventSubscriptions(subs: NotificationSubscription[]) {
+		if (this._isConnected) {
+			const eventTypes = subs.map(x => x.ChainEventType).filter(x => !!x);
+			console.log('Adding Websocket subscriptions for:', eventTypes)
+			this.ceNs.emit('newSubscriptions', eventTypes);
+		} else {
+			console.log('ChainEventsNamespace is not connected');
+		}
+	}
+
+	public deleteChainEventSubscriptions(subs: NotificationSubscription[]) {
+		if (this._isConnected) {
+			const eventTypes = subs.map(x => x.ChainEventType).filter(x => !!x);
+			console.log('Deleting Websocket subscriptions for:', eventTypes)
+			this.ceNs.emit('deleteSubscriptions', subs.map(x => x.ChainEventType));
+		} else {
+			console.log('ChainEventsNamespace is not connected');
+		}
+	}
+
+	private onChainEvent(notification: any) {
```

Comment:
This is great, because what this allows us to do (as a separate PR, ideally) is remove the Chain Events subscriber code from e.g. `client/scripts/controllers/chain/ethereum/compound/governance.ts`, reducing the ETH subscription usage on the frontend, as we should be receiving the related events via the websocket.

Reply:
Indeed, we can listen to all eth events and just funnel them to users accordingly. I actually wasn't aware that we had event subscribers on the front-end as well so this will definitely reduce subscription usage by A LOT.